shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1021,6 +1021,70 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
         val desug = If(IfThen(lhs, rhs), S(Var("false")))
         term.desugaredTerm = S(desug)
         typeTerm(desug)
+      case App(f: Term, a @ Tup(fields)) if (fields.exists(x => x._1.isDefined)) =>
+        val f_ty = typeTerm(f)
+        val fun_ty: SimpleType = f_ty.unwrapProxies match {
+          case tv: TypeVariable =>
+            def getLowerboundFuns(tv: TypeVariable): List[FunctionType] = {
+              val res: List[FunctionType] = tv.lowerBounds.map(x => 
+                x.unwrapProvs match {
+                  case PolymorphicType(_, AliasOf(fun_ty @ FunctionType(_, _))) =>
+                    List(fun_ty)
+                  case tvv: TypeVariable =>
+                    getLowerboundFuns(tvv) 
+                  case _ =>
+                    err("match error", f.toLoc)
```

Comment:
Why is this **user-facing error** reported? First, it doesn't make sense to report an error if one of the lower bounds happens not to be a function; second, the error message is completely meaningless!